### ISSUE TITLE
Feature Request: Pause playback when machine is locked

### ISSUE THREAD

**Issue by eriqs** [MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-#2687]
**OS:**
Windows 10

**Issue Descriptions:**
Feature Request: Pause playback when machine is locked

**Steps to Reproduce:**
I love the app, thank you!  The one thing I don't like is that it doesn't have the option to pause playback when my computer is locked, like other similar apps I have used.  I use this at work, and always lock my computer when I walk away so I would like to have the music pause when I do this, and resume when I return - rather than needing to pause and resume it manually.  Thank you again for the excellent app!

**Comment by jostrander** [comment]
There really is no formal API for this in electron as far as I can tell. So we can't reliably add this.

**Comment by eriqs** [comment]
Bummer. Perhaps the other way around then?  As in a hot key that will pause the player and lock the machine simultaneously?

[sent from my mobile device]
On Aug 6, 2017, at 7:07 PM, Jesse Ostrander <notifications@github.com<mailto:notifications@github.com>> wrote:


There really is no formal API for this in electron as far as I can tell. So we can't reliably add this.

**Comment by MarshallOfSound** [comment]
@jostrander It's not that there's no formal way to do it in Electron, I don't think there's a way to do it on all platforms.

I think `macOS` fires a sleep event, which doesn't exactly equal lock, and windows might fire a window message to the top level window we might be able to read, but on linux it's entirely up to the WM what to tell the windows and when 🤔 

Something to offload to a magical native module I feel, I'll take a look in chromium though and see if they've done the hard work for us.

**Comment by flippinger** [comment]
Any way to do this via Windows task scheduler? At least there is a trigger "on workstation lock".

![image](https://user-images.githubusercontent.com/3148313/29188168-9039984c-7e12-11e7-8991-ea8cfe73a86b.png)

**Comment by Kytech** [comment]
This can be implemented in Windows through the System Media Transport Controls, though we'd have to work out the double skip issue that is currently unresolved. (See #2253)

**Comment by JasonLG1979** [comment]
On Linux it can be accomplished over DBus.

**Comment by liamdawson** [comment]
> I think macOS fires a sleep event

High Sierra supports locking without sleep, so that method won't work in all cases.

**Comment by IndefiniteBen** [comment]
@Kytech Yes please. Music control during connected standby is just about the only thing making me wish for a native Windows store version of GPM.

**Comment by tuur29** [comment]
I made a very quick nodeJS script that uses the Playback API: https://github.com/tuur29/scripts/blob/master/pauseGPMDP.js

**Comment by JasonLG1979** [comment]
I would think the builtin [powerMonitor](https://electronjs.org/docs/api/power-monitor) electron API would be what you'd want to use primarily although for Linux as powerMonitor does not appear to support screen lock events so you'd still need DBus for that.

**Comment by Kytech** [comment]
I know that Windows 10 has a native media transport API that integrates with the lock screen and it already has some integration to GPMDP already. You have to Enable the Windows 10 Media Integration in settings. This will make media controls available on the lock screen in Windows 10.
